compiler/plc_driver/src/pipelines.rs

```diff
@@ -249,7 +294,14 @@ impl<T: SourceContainer + Sync> AnnotatedProject<T> {
             .units
             .iter()
             .map(|(unit, dependencies, literals)| {
-                self.generate_module(context, compile_options, unit, dependencies, literals)
+                self.generate_module(
+                    context,
+                    compile_options,
+                    unit,
+                    dependencies,
+                    literals,
+                    todo!("give GOT layout for single modules?"),
```

Comment:
This causes failures for certain tests, but I am not sure on how it should be handled yet. We could probably pass a custom GOT hashmap here as well, I just can't really work out when `codegen_single_module` or `codegen_to_string` would be used over the current pipeline. I assume `codegen_to_string` is for emitting LLVM IR? If so, we can probably do the same thing as the regular pipeline

Reply:
I handles this the same as default pipeline for now. Usually when we compile a single file, we still need to know about all other symbols at least relevant to this file.